packages/core/src/system.ts

```diff
@@ -1,22 +1,182 @@
 import {
   AnyEventObject,
-  ActorSystem,
   ActorSystemInfo,
   AnyActorRef,
   Observer,
-  Snapshot
+  Snapshot,
+  HomomorphicOmit,
+  EventObject
 } from './types.ts';
 
+export interface ScheduledEvent {
+  id: string;
+  event: EventObject;
+  startedAt: number; // timestamp
+  delay: number;
+  source: AnyActorRef;
+  target: AnyActorRef;
+}
+
+export interface Clock {
+  setTimeout(fn: (...args: any[]) => void, timeout: number): any;
+  clearTimeout(id: any): void;
+}
+
+export interface Scheduler {
+  schedule(
+    source: AnyActorRef,
+    target: AnyActorRef,
+    event: EventObject,
+    data: {
+      id: string | undefined;
+      delay: number;
+    }
+  ): void;
+  cancel(source: AnyActorRef, id: string): void;
+  cancelAll(actorRef: AnyActorRef): void;
+}
+
+export type ScheduledEventId = string & { __scheduledEventId: never };
+
+function getSystemPath(actorRef: AnyActorRef): string {
+  let systemPath = actorRef.id;
+
+  let ancestor = actorRef._parent;
+
+  while (ancestor) {
+    systemPath = ancestor.id + '.' + systemPath;
+    ancestor = ancestor._parent;
+  }
+
+  return systemPath;
+}
+
+function createScheduledEventId(
+  actorRef: AnyActorRef,
+  id: string
+): ScheduledEventId {
+  return `${getSystemPath(actorRef)}.${id}` as ScheduledEventId;
```

Comment:
Couldn't this be simplified like this?
```suggestion
  return `${actorRef.sessionId}.${id}` as ScheduledEventId;
```